### ISSUE TITLE
Color contrast function [NDS-20]

### ISSUE THREAD

**Issue by sh0ji** [wwnorton/design-system#133]
Our books currently use [Bootstrap's `color-yiq` function](https://github.com/twbs/bootstrap/blob/v4.5.0/scss/_functions.scss#L69-L82) to set automatic text color on background-color bleeds, but we cannot currently support that in the design system. Let's do some research on this and implement something that can be used to accomplish it.

## References

- Here's how Bootstrap is handling this now: https://github.com/twbs/bootstrap/blob/2cbb0a941c0a33fc11518061cfd6fad5c0d1c3c0/scss/_functions.scss#L95-L145.
- Could we implement the newer [APCA algorithm](https://www.myndex.com/APCA/), which is forthcoming in WCAG 3.0? If we can, what would be the risk of using this now?

[NDS-20]

[NDS-20]: https://wwnorton.atlassian.net/browse/NDS-20

**Comment by sh0ji** [comment]
This will probably be helpful in #154 and #153.

**Comment by dmedjordan** [comment]
Issue linked to JIRA Story: [NDS-20]

**Comment by Myndex** [comment]
Hi  @sh0ji 

Just wanted to mention that if there are concerns regarding backwards compatibility to WCAG 2 contrast, we've released Bridge-PCA, which is backwards compatible to WCAG 2, but using APCA technology. There's a simple demonstrator and how-to at https://www.myndex.com/BPCA/
 
And both BridgePCA and APCA W3 are available on NPM.

     npm i bridge-pca

     npm i apca-w3

Questions and comments welcome in the discussion forum: https://github.com/Myndex/SAPC-APCA/discussions

**Comment by sh0ji** [comment]
Very cool, thanks for sharing, @Myndex! And thanks for everything you're doing to improve the way we evaluate color. We'll look into using these libraries.
